metadata_fetcher/fetchers/Fetcher.py

```diff
@@ -25,58 +21,32 @@ class FetchError(Exception):
 
 
 class Fetcher(object):
-    def __init__(self, params):
+    def __init__(self, params: dict):
+        """
+        params: dict
+            harvest_type: str
+            collection_id: str or int
+            write_page: str or int filename of the page to write to
+            vernacular_version: path relative to collection id
+                ex: "3433/vernacular_version_1"
+        """
         self.harvest_type = params.get('harvest_type')
         self.collection_id = params.get('collection_id')
         self.write_page = params.get('write_page', 0)
-        bucket = settings.DATA_DEST["BUCKET"]
+        self.vernacular_version = params['vernacular_version']
```

Comment:
Hmm, to your point about a nice error for a missing vernacular version, the only place where I see the fetcher class getting instantiated is here: https://github.com/ucldc/rikolti/blob/rikolti_versioning/metadata_fetcher/lambda_function.py#L42. The vernacular version is programmatically created a few lines up, rather than the result of user input. If there's some sort of failure in the creation of the vernacular version, then this _should_ result in developer intervention, imo. Unlike with the collection id, there's no place for the person running the fetcher to go provide the vernacular version. 

I could add the capacity for a user to specify a vernacular version, I suppose, but that sounds like a whole new feature request with a bunch of added complexity regarding how a harvesting operator might supply a version name.